src/syscall_mock/mod.rs

```diff
@@ -133,26 +136,45 @@ impl SyscallMock {
         Ok(path)
     }
 
-    pub fn handle_end(mut self, exitcode: i32) -> TestResult {
+    pub fn handle_end(mut self, exitcode: i32, captured_stderr: Option<Vec<u8>>) -> TestResult {
         if let Some(expected_step) = self.protocol.steps.pop_front() {
-            self.register_error(&expected_step.command.format(), "<script terminated>");
+            self.register_step_error(&expected_step.command.format(), "<script terminated>");
         }
         if exitcode != self.protocol.exitcode {
-            self.register_error(
+            self.register_step_error(
                 &format!("<exitcode {}>", self.protocol.exitcode),
                 &format!("<exitcode {}>", exitcode),
             );
         }
+        if let Some(expected_stderr) = &self.protocol.stderr {
+            match captured_stderr {
+                None => panic!("stderr expected, but not captured"),
+                Some(captured_stderr) => {
+                    if &captured_stderr != expected_stderr {
+                        self.register_error(format!(
+                            "  expected output to stderr: {:?}\
+                             \n  received output to stderr: {:?}\n",
+                            String::from_utf8_lossy(&expected_stderr.clone()).as_ref(),
+                            String::from_utf8_lossy(&captured_stderr.clone()).as_ref(),
```

Comment:
is `&captured_stderr.clone()` redundant? or is there some type system stuff (`Cow` perhaps?) preventing that?

Reply:
Yes, good call.